SyntheticGeometry/ProjectiveSpace/ConstancyOfFunctions.lagda.md

```diff
@@ -0,0 +1,146 @@
+Constancy of functions ℙⁿ → k
+=============================
+We show that all functions ℙⁿ → k are constant,
+assuming that all functions ℙ¹ → k are constant
+(which we did not formalize yet).
+
+```agda
+{-# OPTIONS --safe #-}
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Structure
+open import Cubical.Foundations.Function
+
+import Cubical.HITs.PropositionalTruncation as PT
+open import Cubical.Data.Nat as ℕ using (ℕ; _+_)
+open import Cubical.Data.FinData
+
+open import Cubical.Algebra.CommRing
+open import Cubical.Algebra.CommRing.LocalRing
+
+open import Cubical.Relation.Nullary.Base using (¬_; yes; no)
+
+import SyntheticGeometry.SQC
+
+module SyntheticGeometry.ProjectiveSpace.ConstancyOfFunctions
+  {ℓ : Level}
+  (k : CommRing ℓ)
+  (k-local : isLocal k)
+  (k-sqc : SyntheticGeometry.SQC.sqc-over-itself k)
+  where
+
+open import SyntheticGeometry.ProjectiveSpace k k-local k-sqc
+open import SyntheticGeometry.ProjectiveSpace.StandardPoints k k-local k-sqc
+open import SyntheticGeometry.ProjectiveSpace.P0 k k-local k-sqc
+open import SyntheticGeometry.ProjectiveSpace.LineThroughPoints k k-local k-sqc
+```
+
+The case n = 0 is simple.
+
+```agda
+module n=0-Case
+  where
+
+  all-functions-constant :
+    (f : ℙ 0 → ⟨ k ⟩) →
+    2-Constant f
+  all-functions-constant f p q = cong f (isContr→isProp isContr-ℙ⁰ p q)
+```
+
+Let us now deduce the case n ≥ 1 from the case n = 1.
+
+```agda
+equal-on-distinct-points :
+  ((f : ℙ 1 → ⟨ k ⟩) → 2-Constant f) →
+  (n : ℕ) →
+  (f : ℙ n → ⟨ k ⟩) →
+  (p q : ℙ n) →
+  ¬ (p ≡ q) →
+  f p ≡ f q
+equal-on-distinct-points ℙ¹-case n f p q p≢q =
+  PT.rec
+    (is-set _ _)
+    (λ (g , hits-p , hits-q) →
+      f p                ≡⟨ sym (cong f hits-p) ⟩
+      f (g (ℙ¹.p zero))  ≡⟨ ℙ¹-case (f ∘ g) _ _ ⟩
+      f (g (ℙ¹.p one))   ≡⟨ cong f hits-q ⟩
+      f q                ∎)
+    (line-through-points-exists p q p≢q)
+  where
+  open CommRingStr (str k) using (is-set)
+  module ℙ¹ = StandardPoints {n = 1}
+
+module n≥1-Case
+  (n-1 : ℕ)
+  where
+
+  n = ℕ.suc n-1
+
+  -- We need to transport some things along the non-definitional equality
+  -- suc n ≡ n + 1, because our definition of ℙ n uses FinVec ⟨ k ⟩ (n + 1).
+
```

Comment:
Should we change our definition instead or was there a good reason to use ```n+1```?

Reply:
Seems to work: #29

Reply:
Yes, I think we should switch to `1 + n` (or `suc n`).